surefire-provider/src/test/java/org/arquillian/smart/testing/surefire/provider/TestExecutionPlannerLoaderTest.java

```diff
@@ -0,0 +1,84 @@
+package org.arquillian.smart.testing.surefire.provider;
+
+import java.io.File;
+import java.util.Collections;
+import org.arquillian.smart.testing.spi.TestExecutionPlanner;
+import org.arquillian.smart.testing.spi.TestExecutionPlannerFactory;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class TestExecutionPlannerLoaderTest {
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void should_find_matching_strategy() throws Exception {
+        // given
+        final JavaSPILoader mockedSpiLoader = mock(JavaSPILoader.class);
+        when(mockedSpiLoader.load(eq(TestExecutionPlannerFactory.class))).thenAnswer(i -> Collections.singletonList(
+            new DummyTestExecutionPlannerFactory()));
+        final TestExecutionPlannerLoader testExecutionPlannerLoader =
+            new TestExecutionPlannerLoader(mockedSpiLoader, new String[] {});
+
+        // when
+        final TestExecutionPlanner exceptionShouldBeThrown = testExecutionPlannerLoader.getPlannerForStrategy("dummy");
```

Comment:
Isn't it obvious? I'm following best practices such as Copy-Paste Design Pattern! Will fix it soon. Thank you